### DISCUSSION TITLE
GROD calculations/ Error using phaseList/checkSinglePhase

### DISCUSSION THREAD

**Discussion by Phade04** [mtex-toolbox/mtex#1228] [Ask Anything] [answered]
Hello,

I'm currently trying to calculate the misorientation angle of the grain reference orientation deviation of EBSD data containing 2 phases.

I'm following the documentation from https://mtex-toolbox.github.io/EBSDGROD.html. However after plotting the misorientation angle of the GROD I'm not able to overlay the subgrain boundaries using following section of the code:

"plot(grains.innerBoundary,'edgeAlpha',grains.innerBoundary.misorientation.angle / (5*degree))".

Everything else is running fine.

I then always get the following error: 

"Error using phaseList/checkSinglePhase (line 307)

----------------------------------------------------------------
Your variable contains the phases: Austenit (Alpha), Martensit (Gamma)

However, you are executing a command that is only permitted for a single phase!

Please read the chapter "select EBSD data" for how to restrict EBSD data or grains to a single phase.
----------------------------------------------------------------

Error in phaseList/get.CS (line 154)
      id = checkSinglePhase(pL);

Error in grainBoundary/get.misorientation (line 155)
      mori = orientation(gB.misrotation,gB.CS{:});

Error in grainBoundary/subsref (line 49)
  [varargout{1:nargout}] = builtin('subsref',gB,s);

Error in grain2d/subsref (line 44)
  [varargout{1:nargout}] = builtin('subsref',grains,s); "

Any help on how to select a single phase is highly appreciated.

Kind regards.

**Comment by ralfHielscher** [comment, marked as answer]
Hi,

the error most likely occurs in the command
``` matlab
grains.innerBoundary.misorientation
```
This will not work in your case as you have different phases. You need to treat the different phase to phase boundaries separately, i.e.,
```matlab
plot(grains.innerBoundary('Austenite','Martensite'),'edgeAlpha',...
   grains.innerBoundary('Austenite','Martensite').misorientation.angle / (5*degree))
plot(grains.innerBoundary('Martensite','Martensite'),'edgeAlpha',...
    grains.innerBoundary('Martensite','Martensite').misorientation.angle / (5*degree))
plot(grains.innerBoundary('Austenite','Austenite'),'edgeAlpha',...
   grains.innerBoundary('Austenite','Austenite').misorientation.angle / (5*degree))
```
 Ralf.

**Reply by Phade04** [reply]
Hello Ralf,

thank you so much for your quick response ! 

I tried your suggestion yesterday and it works just fine.

Much appreciated, 

Phade
